### ISSUE TITLE
[Bug]: Certain tags not in the tag list

### ISSUE THREAD

**Issue by abonhote** [fibelatti/pinboard-kotlin#348]
### Describe the problem

I have two bookmarks with the "hotel" tag. The tag shows up in the bookmark list, but not in the tag list. There are many other tags with the same behaviour, I guess you can check that using your own bookmark list.

Bookmark with the "hotel" tag:
![Screenshot_20241022_081009_Pinkt](https://github.com/user-attachments/assets/afc4c85b-337d-4b5b-bbb0-d9c05b486dc8)

Tag list:
![Screenshot_20241022_081049_Pinkt](https://github.com/user-attachments/assets/c79ebdf7-3d93-4cb1-9d80-151456b4e3fb)

### Device model

Samsung Galaxy S23 Ultra

### Android version

Android 14

### App version

3.3.0

**Comment by fibelatti** [comment]
Hi @abonhote, thanks for submitting this issue. I haven't faced this myself, but let me investigate to see what's going on. In the meantime, I have two questions:

1. Are you using Pinboard or Linkding?
2. Does the tag appear in the tag list whenever you are using the web interface?

**Comment by abonhote** [comment]
Hey!

1 linkding
2 yes

**Comment by fibelatti** [comment]
Alright, thanks for confirming those. I'll investigate the issue and fix it as soon as possible.

**Comment by abonhote** [comment]
Were you able to reproduce it? Which API endpoint are you using? I could run a test on my bookmarks, just to figure out whether it's the api or pinkt.

**Comment by fibelatti** [comment]
@abonhote I haven't had the chance to look at this just yet as I had to fix an issue with build distribution.

The app uses `GET /api/tags/` to fetch tags. It would be great if you could check whether hitting the API directly behaves as expected, because if it does then we'll know the issue is definitely the app.

**Comment by abonhote** [comment]
Hi!

Querying the API directly works here. I have 163 tags, that specific one is on page two (limit=100 is default). In fact, all tags on the second result page are not visible in pinkt, so I guess that's the pattern 😉

https://linkding.link/api/#tags

When I GET `/api/tags/` without any param, my JSON looks like this:

```json
{
	"count": 163,
	"next": "http://xxx/api/tags/?limit=100&offset=100",
	"previous": null,
	"results":[
		{
		"id": 32,
		"name": "tag_1",
		"date_added": "2024-07-13T14:42:38.152257Z"
		}
	]
}
```

Note the `next` value; if there is one, you'll need to fetch that one too, until it's null.

Cheers

André

**Comment by abonhote** [comment]
You'll probably want to replace the part of the URL up to `/api/` with your user's settings. Here, I get the URL behind the reverse proxy, no https. That would not work.

**Comment by fibelatti** [comment]
> In fact, all tags on the second result page are not visible in pinkt, so I guess that's the pattern 😉

@abonhote good finding! Thanks for checking that, I thought I was already handling pagination the same way I do for bookmarks but that's clearly not the case. Pinboard doesn't have it, so no wonder I missed it when implementing the Linkding variant.

I'll get the fixed in the next couple days and release a patched version.

**Comment by fibelatti** [comment]
@abonhote this issue has been fixed in version v3.3.3.

Thanks for reporting it, if the issue persists please feel free to reopen.

**Comment by abonhote** [comment]
Works perfectly fine now, thanks!
